app/src/main/java/io/gnosis/safe/helpers/ConnectivityInfoProvider.kt

```diff
@@ -18,7 +18,7 @@ class ConnectivityInfoProvider(private val connectivityManager: ConnectivityMana
 
         override fun onLost(network: Network?) {
             super.onLost(network)
-            offline = true
+            offline = connectivityManager.activeNetworkInfo == null
```

Comment:
> Do NOT call ConnectivityManager.getNetworkCapabilities(android.net.Network) or ConnectivityManager.getLinkProperties(android.net.Network) or other synchronous ConnectivityManager methods in this callback as this is prone to race conditions ; calling these methods while in a callback may return an outdated or even a null object.

from https://developer.android.com/reference/android/net/ConnectivityManager.NetworkCallback#onLost(android.net.Network) 😄

Reply:
If @biafra23 says it works, I am fine with the solution ... for me it was more about what happens if 2 networks are active at the same time (is that even possible?) ... since this method is used to block all network traffic of the app we need to be sure how this works.